src/query/management/src/role/role_mgr.rs

```diff
@@ -158,7 +157,13 @@ impl RoleApi for RoleMgr {
             res.ok_or_else(|| ErrorCode::UnknownRole(format!("Role '{}' does not exist.", role)))?;
 
         match seq.match_seq(&seq_value) {
-            Ok(_) => Ok(seq_value.into_seqv()?),
+            Ok(_) => {
+                let data = serde_json::from_slice::<RoleInfo>(&seq_value.data).or_else(|err| {
+                    log::debug!("serde json err when get role. cause : {}", err);
```

Comment:
I would prefer to `deserialize_struct` first rather than `serde_json::from_slice` here.
Reson:
If `cdeserialize_struct` first, we can know a version is ok. `serde_json::from_slice` only affects some version, but the future will be `cdeserialize_struct`.
cc @drmingdrmer

Reply:
I'm totally fine with running `deserialize_struct` right before `serde_json`. :)

Reply:
@TCeason what do you think?

Reply:
That's ok. But better modify it in next pr with serialize pb?

In this pr the role info also serialize as json. So no need to deserialize first.

Reply:
Ok, we need to upgrade the json format to the new proto when reading(fix on reading), or the json format will not changing for ever.

Reply:
> Ok, we need to upgrade the json format to the new proto when reading(fix on reading), or the json format will not changing for ever.

I think No need upgrade?

In if user grant or revoke the wold role, it will call upsert role.

In this method, will call get role， and then serialize the new privileges into role.